File: 2610-closest-prime-numbers-in-range/closest-prime-numbers-in-range.py
class Solution:
    def closestPrimes(self, left: int, right: int) -> List[int]:


        maxval = int(right**0.5)+1
        isPrime = (right+1) * [True]
        isPrime[0] = False
        isPrime[1] = False
        for i in range(2, maxval):
            if isPrime[i]:
                j = 2*i
                while j <= right:
                    isPrime[j] = False
                    j += i

        prev = None
        ans = [-1,-1]
        minval = inf
        for i in range(left,right+1):
            if isPrime[i]:
                if prev and i-prev < minval:
                    minval = i-prev
                    ans =[prev,i]
                prev = i

        return ans

        return ans
        

File: 2610-closest-prime-numbers-in-range/closest-prime-numbers-in-range.cpp
class Solution {
public:
    vector<int> closestPrimes(int left, int right) {

        vector<bool> isPrime(right+1, true);
        vector<int> ans = {-1,-1};
        isPrime[0] = false;
        isPrime[1] = false;
        int maxval = int(sqrt(right)) + 1;

        for (int i = 2; i <= maxval; ++i) {
            if (isPrime[i]) {
                for (int j = 2*i; j <= right; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        int prev = -1;
        int minval = 1e9;
        for (int i = left; i <= right; ++i) {
            if (isPrime[i]) {
                if (prev > 0 and (i - prev < minval)) {
                    ans = {prev, i};
                    minval = i - prev;
                }
                prev = i;
            }
        }

        return ans;
        
    }
};